src/dodal/devices/eiger.py

```diff
@@ -158,23 +162,28 @@ def unstage(self) -> bool:
                 self.timeouts.general_status_timeout
             )
             self.disable_roi_mode()
+        self.disarming_status.set_finished()
         return status_ok
 
     def stop(self, *args):
         """Emergency stop the device, mainly used to clean up after error."""
         LOGGER.info("Eiger stop() called - cleaning up...")
-        self.wait_on_arming_if_started()
-        stop_status = self.odin.stop()
-        self.odin.file_writer.start_timeout.set(1).wait(
-            self.timeouts.general_status_timeout
-        )
-        self.disarm_detector()
-        stop_status &= self.disable_roi_mode()
-        stop_status.wait(self.timeouts.general_status_timeout)
-        # See https://github.com/DiamondLightSource/hyperion/issues/1395
-        LOGGER.info("Turning off Eiger dev/shm streaming")
-        self.odin.fan.dev_shm_enable.set(0).wait()
-        LOGGER.info("Eiger has successfully been stopped")
+        if not self.disarming_status.done:
+            LOGGER.info("Eiger still disarming, waiting on disarm")
+            self.disarming_status.wait(self.timeouts.arming_timeout)
+        else:
```

Comment:
Perhaps this fixes the issue, but I'm unclear on what exactly we should do in `stop()`, looking at bluesky `run_engine.py` it seems `stop()` is invoked in a variety of scenarios, including at the end of a plan, during a suspend and when `pause()` is invoked. The impression I get is that being stopped is at least somewhat resumable.